src/fileWatcher.ts

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+import { Disposable, ExtensionContext, FileSystemWatcher, workspace, Uri } from "vscode";
+import { LanguageClient, TextDocumentIdentifier } from "vscode-languageclient";
+
+export class FileWatcher {
+    private watcher: FileSystemWatcher;
+    private onDeleteHandlers: Array<Function>;
+    private onCreateHandlers: Array<Function>;
+
+    public constructor(private context: ExtensionContext, private clients: Map<string, LanguageClient>) {
+        this.watcher = workspace.createFileSystemWatcher("**/*");
```

Comment:
this wildcard will cause the filewatcher events to be fired for irrelevant files too, ex. log/node_modules/etc.

Reply:
@sriv you are right about that, but we want to watch for implementation files too here. Knowing the language of each file is not possible from VSCode if the file is not opened or deleted. Hence, I had to put a wildcard here. Do you have a better approach for this?

Reply:
if the runner honours implementation dir env, then we can use that and https://github.com/getgauge/gauge/pull/985 will enable us to use the env to locate the spec folder.

If not, then I guess we can blacklist some files rather than whitelist. i.e. use the [GlobPattern](https://code.visualstudio.com/docs/extensionAPI/vscode-api#GlobPattern) to ignore watching the below files/folders:

- .git
- .gauge
- logs
- bin
- gauge_bin
- reports
- node_modules (javascript)
- vendor (ruby bundler)
- packages (c# nuget)
- libs